nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/ClusterProtocolHeartbeatMonitor.java

```diff
@@ -91,12 +91,8 @@ public ClusterProtocolHeartbeatMonitor(final ClusterCoordinator clusterCoordinat
         this.clusterNodesPath = zkClientConfig.resolvePath("cluster/nodes");
 
         String hostname = properties.getProperty(NiFiProperties.CLUSTER_NODE_ADDRESS);
-        if (hostname == null) {
-            try {
-                hostname = InetAddress.getLocalHost().getHostName();
-            } catch (UnknownHostException e) {
-                throw new RuntimeException("Unable to determine local hostname and the '" + NiFiProperties.CLUSTER_NODE_ADDRESS + "' property is not set");
-            }
+        if (hostname == null || hostname.trim().isEmpty()) {
+            hostname = "localhost";
```

Comment:
Thanks for the explanation. I found that [docs says so](https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#clustering).

I have read `nifi.remote.input.socket.host` description before, and wondered what's the reason of these difference:
```
By default, it is the value from InetAddress.getLocalHost().getHostName(). On UNIX-like operating systems, this is typically the output from the hostname command.
```

I understand that as to keep how it was done in 0.x.